# Exploratory Data Analysis

---

## Importing Libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

## Loading Dataset

In [5]:
epl_df = pd.read_csv('D:\git_repositories\PythonProject2-Exploratory_Data_Analysis_Using_Python\Data\EPL_20_21.csv')

In [6]:
epl_df.head()

Name     Club Nationality Position  Age  Matches  Starts  Mins  \
0    Mason Mount  Chelsea         ENG    MF,FW   21       36      32  2890   
1  Edouard Mendy  Chelsea         SEN       GK   28       31      31  2745   
2    Timo Werner  Chelsea         GER       FW   24       35      29  2602   
3   Ben Chilwell  Chelsea         ENG       DF   23       27      27  2286   
4    Reece James  Chelsea         ENG       DF   20       32      25  2373   

   Goals  Assists  Passes_Attempted  Perc_Passes_Completed  Penalty_Goals  \
0      6        5              1881                   82.3              1   
1      0        0              1007                   84.6              0   
2      6        8               826                   77.2              0   
3      3        5              1806                   78.6              0   
4      1        2              1987                   85.0              0   

   Penalty_Attempted    xG    xA  Yellow_Cards  Red_Cards  
0                  1  0.21  0.24             2          0  
1                  0  0.00  0.00             2          0  
2                  0  0.41  0.21             2          0  
3                  0  0.10  0.11             3          0  
4                  0  0.06  0.12             3          0

**Information about the dataset**

In [7]:
epl_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 532 entries, 0 to 531
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Name                   532 non-null    object 
 1   Club                   532 non-null    object 
 2   Nationality            532 non-null    object 
 3   Position               532 non-null    object 
 4   Age                    532 non-null    int64  
 5   Matches                532 non-null    int64  
 6   Starts                 532 non-null    int64  
 7   Mins                   532 non-null    int64  
 8   Goals                  532 non-null    int64  
 9   Assists                532 non-null    int64  
 10  Passes_Attempted       532 non-null    int64  
 11  Perc_Passes_Completed  532 non-null    float64
 12  Penalty_Goals          532 non-null    int64  
 13  Penalty_Attempted      532 non-null    int64  
 14  xG                     532 non-null    float64
 15  xA    

In this English Premier League (2020-2021) dataset, we can see that there are 18 columns, 532 rows and no missing values.